src/Microsoft.AspNet.Diagnostics/Views/BaseView.cs

```diff
@@ -46,6 +52,7 @@ public abstract class BaseView
             Request = Context.Request;
             Response = Context.Response;
             Output = new StreamWriter(Response.Body);
+            HtmlEncoder = context.ApplicationServices.GetRequiredService<IHtmlEncoder>();
```

Comment:
Consider the ApplicationServices.GetHtmlEncoder() extension method, as it will return a default instance if one hasn't been registered with DI.